src/ngraph_encapsulate_op.cc

```diff
@@ -543,9 +567,23 @@ class NGraphEncapsulateOp : public OpKernel {
         ng_inputs.push_back(nullptr);
         continue;
       }
-
       NGRAPH_VLOG(4) << "NGraphEncapsulateOp:: Input from non Variable Node";
 #endif
+
+      bool has_data_input =
+          NGraphCatalog::ExistsInInputDataTensorMap(def().input(i));
+      if (has_data_input) {
+        NGRAPH_VLOG(4) << " Input data already in device. For Input Index " << i
+                       << endl;
+
+        auto data_input_tensor =
+            NGraphCatalog::GetTensorFromInputDataTensorMap(def().input(i));
+        OP_REQUIRES(ctx, data_input_tensor != nullptr,
+                    errors::Internal("Not found input data on Device.\n"));
+        ng_inputs.push_back(data_input_tensor);
+        continue;
```

Comment:
i'd prefer relying in if-else than in if-continue. if-else makes it easier to read i think. it makes it clear there is only 2 ways to get data... from`GetTensorFromInputDataTensorMap` or the usual way `GetCurrentNgTensor`

Reply:
There are 3 ways right now as we also have the if for Variables also, but it is in if-def. I can put this in if-else and let that one (Variable) be as it is, with continue.

Reply:
Added comments for now